src/nfa/nfa.cc

```diff
@@ -308,6 +317,18 @@ void Nfa::remove_epsilon(const Symbol epsilon)
     *this = Mata::Nfa::remove_epsilon(*this, epsilon);
 }
 
+TransitionList::const_iterator Nfa::get_transitions_from_state_with(State state_from, Symbol symbol) const {
+    const auto state_transitions_iter_end{ transitionrelation[state_from].end() };
+    for (auto state_transitions_iter{ transitionrelation[state_from].begin() };
```

Comment:
Cannot this be done without the .begin() and .end()? Shouldn't this be automatically be supported as `foreach`?

Reply:
If you mean `std::for_each`, it could be, but I want to return the iterators here to allow simple testing whether the transitions with the given symbol were found. The current implementation probably exactly mirrors the implementation of `std::for_each` without the need to create a lambda function to pass to the `std::for_each`. I would say that this is a better, cleaner solution, but only because I want to use and return the underlaying iterators of `std::for_each`.

Reply:
Fine by me.